src/App.tsx

```diff
@@ -26,70 +27,138 @@ type ReorderOptions = {
+
+export class App extends React.PureComponent<{}, State> {
+  state: Readonly<State> = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  changeSatate(typeMove?: string, typeSort?: string) {
+    if (typeMove === 'sort' && typeSort === 'alpha') {
+      return this.setState((prev: State) => {
+        return {
+          ...prev,
+          sortType: SortType.ALPHABET,
+        };
+      });
+    }
+
+    if (typeMove === 'sort' && typeSort === 'length') {
+      return this.setState((prev: State) => {
+        return {
+          ...prev,
+          sortType: SortType.LENGTH,
+        };
+      });
+    }
+
+    if (typeMove === 'reverse' && !typeSort) {
+      return this.setState((prev: State) => {
+        return {
+          ...prev,
+          isReversed: !prev.isReversed,
+        };
+      });
+    }
+
+    return this.setState((prev: State) => {
+      return {
+        ...prev,
+        sortType: SortType.NONE,
+        isReversed: false,
+      };
+    });
+  }
+
+  render(): React.ReactNode {
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            onClick={() => this.changeSatate('sort', 'alpha')}
+            className={
+              classNames('button is-info', {
+                'is-light': this.state.sortType !== SortType.ALPHABET,
+              })
+            }
+            type="button"
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            onClick={() => this.changeSatate('sort', 'length')}
+            className={
+              classNames('button is-info', {
+                'is-light': this.state.sortType !== SortType.LENGTH,
+              })
+            }
+            type="button"
+          >
+            Sort by length
+          </button>
+
+          <button
+            onClick={() => this.changeSatate('reverse')}
+            className={
+              classNames('button is-warning', {
+                'is-light': !this.state.isReversed,
+              })
+            }
+            type="button"
+          >
+            Reverse
+          </button>
+
+          {this.state.sortType || this.state.isReversed ? (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => this.changeSatate()}
+            >
+              Reset
+            </button>
+          ) : ''}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {
+            getReorderedGoods(goodsFromServer, this.state).map(good => (
+              <li
+                key={good}
+                data-cy="Good"
+              >
+                {good}
+              </li>
+            ))
+          }
```

Comment:
it would be better to change this code for better readability
```suggestion
          {getReorderedGoods(goodsFromServer, this.state).map(good => (
            <li key={good} data-cy="Good">
              {good}
            </li>
          ))}
```